packages/core/core/src/services/document-service/utils/unidirectional-relations.ts

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable no-continue */
+import { UID, Schema } from '@strapi/types';
+import { async, traverseEntity } from '@strapi/utils';
+
+interface RelationToSync {
+  uid: string;
+  attribute: string;
+}
+
+/**
+ * Updates uni directional relations to target the right entries when overriding published or draft entries.
+ *
+ * 1. Finds all content types that target the content type that's about to be published/discarded
+ * 2. For each relation, update them to target the new entry
+ *
+ * @param oldEntries - The old entries that will be removed.
+ * @param newEntries - The new entries that unidirectional relations should target.
+ */
+const syncUnidirectionalRelations = async (
+  uid: UID.ContentType,
+  oldEntries: { id: string; locale: string }[],
+  newEntries: { id: string; locale: string }[]
+) => {
+  const relationsToSync: RelationToSync[] = [];
+
+  // Iterate all components and content types
+  const contentTypes = Object.values(strapi.contentTypes) as Schema.ContentType[];
+  const components = Object.values(strapi.components) as Schema.Component[];
+
+  for (const contentType of [...contentTypes, ...components]) {
+    // If the content type has a relation to the current content type
+    for (const [name, attribute] of Object.entries(contentType.attributes) as any) {
+      if (attribute.type !== 'relation') continue;
```

Comment:
Would it be safer to target the specific oneWay & manyWay relational types directly ?

Reply:
I don't fully understand the question, isn't this bit only selecting the one way/many way relations ? (by checking if the attr has inversedBy & mappedBy)

Reply:
do you know of other ways of filtering by one way relations? I was checking the available metadata on the attribute and we only store the following:
```
{
  type: "relation",
  relation: "oneToOne",
  target: "api::tag.tag",
}
```

Reply:
Ok that's my bad, we call them oneWay & manyWay only inside the ctb 🤦

Reply:
Trust me I do that too 😂